### PR TITLE
Delete dead Tensor code paths

### PR DESCRIPTION
**Pull request by colesbury**
This deletes most of the dead Tensor code paths, including the TensorMethods cwrap and generic/Tensor.cpp.

This also moves the THNN.cwrap/.cpp generation to generate_code which can use ninja if installed.

### CODE REVIEW

File: torch/csrc/autograd/functions/init.cpp [hunk 1]
```diff
@@ -110,10 +110,11 @@ bool THPAutograd_initFunctions(PyObject* _unused)
 
   generated::initialize_autogenerated_functions();
 
-  THPObjectPtr parent(PyImport_ImportModule("torch._C"));
-  if (!parent) return false;
-  PyModule_AddObject(parent.get(), "_functions", module.release());
```
Comment (gchanan) [inline torch/csrc/autograd/functions/init.cpp]: do you need to incref this?

**Review by gchanan — APPROVED** [review]

File: torch/csrc/autograd/functions/init.cpp (hunk 1 above)
Comment (colesbury) [reply torch/csrc/autograd/functions/init.cpp]: I'll add an explicit `Py_INCREF` to make it more clear, but the `module.release()` releases ownership and PyModule_AddObject steals it.

**Comment by colesbury** [comment]
ASAN build seems flaky today. ONNX build passed earlier

**Comment by sakaia** [comment]
It seems "tools/cwrap/plugins/THPPlugin.py" should be removed, since no other module imported "class THPPlugin(CWrapPlugin):"

**Pull request merged**